### PR TITLE
update checker

### LINKED ISSUE #842: [Feature Req] Cloud function to check for stale data
We will have a cloud function periodically checking the live API to see if data is out of date. In the event it is, we will send a message in a discord channel dedicated to reporting this.
This will in the end be hosted serverless by Greymatter on an AWS Lambda function cron job.
PR#: #849

### PR DESCRIPTION
**Pull request by puf17640**
closes #842

### CODE REVIEW

**Comment by ebwinters** [comment]
A few things
1. We should have some allowance for out of date-ness. Maybe allow for double the interval, due to caching, and also sometimes a source that updates once every 24h posts later, and it turns into 36h for a day or two.
2. We should be sending messages to a discord channel when we find an our of date endpoint. 
3. This is going to end up running in a cloud function.
4. There is no NYT checker, don't know if that's on purpose.

**Comment by puf17640** [comment]
> A few things
> 
> 1. We should have some allowance for out of date-ness. Maybe allow for double the interval, due to caching, and also sometimes a source that updates once every 24h posts later, and it turns into 36h for a day or two.
> 2. We should be sending messages to a discord channel when we find an our of date endpoint.
> 3. This is going to end up running in a cloud function.
> 4. There is no NYT checker, don't know if that's on purpose.

4) there is no timestamp provided, I left out quite a few endpoints because of this, we could discuss adding timestamps to those endpoints but it will maybe kill applications using the endpoint with a fixed model
3) alrighty, i will expose the function later on, this is WIP anyway
2) sure just provide me with a webhook and i will
1) sure, so pass in args? or what would you propose?

**Comment by ebwinters** [comment]
Idk what you mean pass in args, but when you're checking if it's greater than the configured interval, just double the interval to allow for some small out of date entries due to source errors

**Comment by puf17640** [comment]
> Idk what you mean pass in args, but when you're checking if it's greater than the configured interval, just double the interval to allow for some small out of date entries due to source errors 

oh okay I thought you meant I should make it more configurable, alright will do

**Comment by ebwinters** [comment]
Also, lets create a channel for these messages t o be sent, not in the development channel

**Comment by puf17640** [comment]
> Also, lets create a channel for these messages t o be sent, not in the development channel

i was just trying it out

File: updateCheck.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+							title: `${disease}/${endpoint} - Error`,
+							url: `https://disease.sh/v3/${disease}/${endpoint}`,
+							description: err.message || 'No message'
+						}
+					]
+				});
+			}
+		}
+	}
+	process.exit();
```
[the remark this answers is not in the document]
Comment (puf17640) [reply updateCheck.js:74]: because it would not exit out of the program somehow

**Comment by dontlaugh** [comment]
Here is a k8s CronJob I've sketched out
https://gist.github.com/dontlaugh/e2dcc4c933d9d69678e7c795434546eb#file-update-checker-yaml-L6-L8

These flux annotations should watch the same docker repository as your API server, and deploy updates to the code in the same way. The webhook should be set as an env var.

I've added an npm command in there, but that might need to change when you add that back in. Lemme know what it should be :+1:

**Comment by ebwinters** [comment]
So is this good to merge? Is the config set up to run it?

**Comment by dontlaugh** [comment]
Needs an update to package.json, so we can run this as an npm script https://github.com/disease-sh/API/blob/master/package.json#L15

**Pull request merged**